test/units/modules/test_apt.py

```diff
@@ -43,3 +47,60 @@
 def test_expand_pkgspec_from_fnmatches(test_input, expected):
     """Test positive cases of ``expand_pkgspec_from_fnmatches``."""
     assert expand_pkgspec_from_fnmatches(None, test_input, fake_cache) == expected
+
+
+def test_install_deb_filters_virtual_packages():
```

Comment:
I would write an integration test by writing a spec file like the following under - https://github.com/ansible/ansible/tree/devel/test/integration/targets/setup_deb_repo/files/package_specs/stable 

```
Section: misc
Priority: optional
Standards-Version: 2.3.3

Package: packagethree
Version: 1.0
Section: system
Maintainer: John Doe <john@doe.com>
Architecture: all
Description: Dummy package package three
Provides: virtualpackage
```

Also, could you remove this unit test?